### PR TITLE
Support query nested properties

### PR DESCRIPTION
**Pull request by jannyHou**
connect to strongloop/loopback#517

Support query nested properties

### CODE REVIEW

**Comment by loay** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

File: test/basic-querying.test.js [hunk 1]
```diff
@@ -574,13 +590,71 @@ describe('basic-querying', function() {
...
       sample('id').expect(['id']);
       sample(['id']).expect(['id']);
       sample(['email']).expect(['email']);
     });
+
+    var describeWhenNestedSupported = connectorCapabilities.nestedProperty ? describe : describe.skip;
+    describeWhenNestedSupported('query with nested property', function() {
+      it('should support nested property in query', function(done) {
+        User.find({where: {'address.city': 'San Jose'}}, function(err, users) {
+          should.not.exist(err);
```
Comment (loay) [inline test/basic-querying.test.js]: if (err) return done(err);

File: test/basic-querying.test.js [hunk 2]
```diff
@@ -574,13 +590,71 @@ describe('basic-querying', function() {
...
+          for (var i = 0; i < users.length; i++) {
+            users[i].address.city.should.be.eql('San Jose');
+          }
+          done();
+        });
+      });
+
+      it('should support nested property with regex over arrays in query', function(done) {
+        User.find({where: {'friends.name': {regexp: /^Ringo/}}}, function(err, users) {
+          should.not.exist(err);
```
Comment (loay) [inline test/basic-querying.test.js]: ditto

File: test/basic-querying.test.js [hunk 3]
```diff
@@ -574,13 +590,71 @@ describe('basic-querying', function() {
...
+          var expectedUsers = ['John Lennon', 'Paul McCartney'];
+          (expectedUsers.indexOf(users[0].name) > -1).should.be.ok();
+          (expectedUsers.indexOf(users[1].name) > -1).should.be.ok();
+          done();
+        });
+      });
+
+      it('should support nested property with gt in query', function(done) {
+        User.find({where: {'address.city': {gt: 'San'}}}, function(err, users) {
+          should.not.exist(err);
```
Comment (loay) [inline test/basic-querying.test.js]: if (err) return done(err);

File: test/basic-querying.test.js [hunk 4]
```diff
@@ -574,13 +590,71 @@ describe('basic-querying', function() {
...
+            users[i].address.state.should.be.eql('CA');
+          }
+          done();
+        });
+      });
+
+      it('should support nested property for order in query', function(done) {
+        User.find({where: {'address.state': 'CA'}, order: 'address.city DESC'},
+          function(err, users) {
+            should.not.exist(err);
```
Comment (loay) [inline test/basic-querying.test.js]: ditto

File: test/basic-querying.test.js [hunk 5]
```diff
@@ -574,13 +590,71 @@ describe('basic-querying', function() {
...
+            users.length.should.be.equal(2);
+            users[0].address.city.should.be.eql('San Mateo');
+            users[1].address.city.should.be.eql('San Jose');
+            done();
+          });
+      });
+
+      it('should support multi-level nested array property in query', function(done) {
+        User.find({where: {'address.tags.tag': 'business'}}, function(err, users) {
+          should.not.exist(err);
```
Comment (loay) [inline test/basic-querying.test.js]: if (err) return done(err);

**Comment by loay** [comment]
I think the error handler should be done as in my comment, other than that, the pr looks fine

**Comment by jannyHou** [comment]
oracle still fails due to installation problem and test cases in dashdb is waiting for a new release of `ibm_db`
Merging it.

**Pull request merged**